### PR TITLE
Fix GeluRecompute for 2 inputs case.

### PR DESCRIPTION
**Pull request by kit1980**

### CODE REVIEW

File: onnxruntime/test/optimizer/graph_transform_test.cc [hunk 1]
```diff
@@ -2115,6 +2118,35 @@ TEST_F(GraphTransformationTests, BiasGeluTest) {
   ASSERT_TRUE(op_to_count["com.microsoft.BiasGelu"] == 1);
 }
 
+#ifdef ENABLE_TRAINING
```
Comment (SherlockNoMad) [inline onnxruntime/test/optimizer/graph_transform_test.cc:2121]: you can place this under orttraining/test/optimizer/graph_transformer_test.cc

The file is for training only graph transformers

File: orttraining/orttraining/core/optimizer/localized_recompute.cc [hunk 2]
```diff
@@ -36,6 +36,12 @@ Status GeluRecompute::ApplyImpl(Graph& graph, bool& modified, int /*graph_level*
       continue;
     }
 
+    std::vector<NodeArg*> gelu_input = {node.MutableInputDefs()[0]};
```
Comment (SherlockNoMad) [inline orttraining/orttraining/core/optimizer/localized_recompute.cc:39]: Just a for loop to go through all the inputs? 
By default, the recompute node should copy everything from the original node.

Comment (kit1980) [reply orttraining/orttraining/core/optimizer/localized_recompute.cc]: I think we don't even need a loop or copy, now I just pass `MutableInputDefs()` to `AddNode()`.
Any potential issues I'm missing?

File: onnxruntime/test/optimizer/graph_transform_test.cc (hunk 1 above)
Comment (kit1980) [reply onnxruntime/test/optimizer/graph_transform_test.cc]: Done.

File: orttraining/orttraining/core/optimizer/localized_recompute.cc (hunk 2 above)
Comment (SherlockNoMad) [reply orttraining/orttraining/core/optimizer/localized_recompute.cc]: yes. MutableInputDefs() is good enough

**Review by SherlockNoMad — APPROVED** [review]

**Review by SherlockNoMad — APPROVED** [review]

**Pull request merged**